src/docs/development/platform-integration/platform-channels.md

```diff
@@ -637,6 +636,52 @@ If using the iOS Simulator,
 note that it does not support battery APIs,
 and the app displays 'battery info unavailable'.
 
+## Typesafe platform channels via Pigeon {#pigeon}
+
+The previous example uses `MethodChannel` to communicate between the host and
+client which isn't typesafe.  Calling and receiving messages depends on the host
+and client declaring the same arguments and datatypes in order for messages to
+work.  The [Pigeon][] package can be used as an alternative to `MethodChannel`
+to generate code that sends messages in a typesafe manner.
+
+With [Pigeon][] the messaging protocol is defined in a subset of Dart which then
+generates messaging code for Android or iOS.  A more complete example and more
+information can be found on the [Pigeon pub.dev page][];
```

Comment:
I think we can list out Pigeon's advantages a bit more. Real method callbacks on a real class, invoke real methods on a real class, widely usable data types, no hand coding strings, built in error handling, built in returns etc.

Reply:
done, added list of benefits